src/services/decoders/ClpIrDecoder.ts

```diff
@@ -60,7 +89,36 @@ class ClpIrDecoder implements Decoder {
         endIdx: number,
         useFilter: boolean
     ): Nullable<DecodeResultType[]> {
-        return this.#streamReader.decodeRange(beginIdx, endIdx, useFilter);
+        const results = this.#streamReader.decodeRange(beginIdx, endIdx, useFilter);
+        if (null === this.#formatter) {
+            return results;
+        }
+        results.forEach((result) => {
+            const [message, timestamp] = result;
+            let fields: JsonObject = {};
+            try {
+                fields = JSON.parse(message) as JsonObject;
+                if (false === isJsonObject(fields)) {
+                    throw new Error("Unexpected non-object.");
+                } else if (null !== this.#logLevelKey) {
+                    const logLevel = fields[this.#logLevelKey];
+                    fields[this.#logLevelKey] = LOG_LEVEL[logLevel as LOG_LEVEL];
+                }
+            } catch (e) {
+                if (0 === message.length) {
+                    return;
+                }
+                console.error(e, message);
+            }
+
+            // @ts-expect-error `this.#formatter` is certainly non-null
+            result[0] = this.#formatter.formatLogEvent({
+                fields: fields,
+                timestamp: convertToDayjsTimestamp(timestamp),
+            });
+        });
```

Comment:
> I feel like we dont need to get the log level since it should already be in the result. 

Right. For now we shall store log level strings as authoritative log levels in the logging library.

> I also dont think we need to check if the length is 0, since i dont think clp-ffi-js would return length 0. I think worst case it will return {} if the log event is empty.

Good catch